storage/replica.go

```diff
@@ -1151,6 +1156,97 @@ func (r *Replica) maybeSetCorrupt(err error) error {
 	return err
 }
 
+// resolveIntents resolves the given intents. Those which are local to the range
```

Comment:
check out the comment in `resolveWriteIntentError`. Basically local path = local intents with clients waiting, "slow" path = remote intents that can be done completely async. The latter occurs only now because `EndTransaction` returns its remote intents as skipped intents via existing mechanisms (which seems wise because the alternative is calling directly from very low-level code to very high-level code). If that code isn't here, it's going to be somewhere else, and having a method that you give a bunch of intents to and it does the right thing seems good, though I imagine we'll refactor it a bit over time. BTW, your recent GC intent stuff could probably just reuse parts here instead of patching together its own requests.